_posts/2018-12-05-exploit-model.md

```diff
@@ -0,0 +1,195 @@
+
+Now, let's take a look at [Max's work with gVisor](https://justi.cz/security/2018/11/14/gVisor-lpe.html). In his blog, he explains the bug that he has found in the gVisor implementation of the `shmctl` syscall, and an example of how he is able to perform targeted writes to a seperate process using the same gVisor userspace kernel.
+
+The exploit assumes the following attacker capabilities: Code exec in an application process. This can be obtained via exploitation of the process `p1a` (i.e. redis/nginx). We annotate the diagram with attacker capabilities assumed in parentheses.
+
+```
++--------+ +--------+   +---------+
+|  p1a   | |  p1b   |   |   p2a   |                      APP
+| (code) | |        |   |         |                      
+| (exec) | |        |   |         |                      
++--------+ +--------+   +---------+
++-------------------+   +-------------------+
+|      gVisor       |   |      gVisor       |            gVisor
++-------------------+   +-------------------+
++--------------------------------------------------+
+|                    kernel                        |     KERNEL
++--------------------------------------------------+
+```
+
+The exploit abuses the `shmctl` bug to acheive the attacker capabilities to perform limited write to select memory region of which gVisor has access to. Max then uses this limited write capability to show that he has obtained capability of random writes to the adjacent application process `p1b`, by attempting to write 'A's (`0x41`) to the specific region/pages exposed by the bug.
+
+```
+    +---WRITE--+
+    |          |
+    |          V
++--------+ +--------+   +---------+
+|  p1a   | |  p1b   |   |   p2a   |                      APP
+| (code) | |(random)|   |         |                      
+| (exec) | |( write)|   |         |                      
++--------+ +--------+   +---------+
++-------------------+   +-------------------+
+|      gVisor       |   |      gVisor       |            gVisor
+|  (limited write)  |   |                   |            
++-------------------+   +-------------------+
++--------------------------------------------------+
+|                    kernel                        |     KERNEL
++--------------------------------------------------+
+```
+
+### Safe for now
+
+This proof of concept has shown random writes into an adjacent process on the same instance of gVisor. However, we note that to get the most out of gVisor isolation, one would want to have each process be on seperate gVisor instances. Thus, there is no reason to panic today, based on this proof of concept alone.
+
+### Other implications
+
+This is one of many direction that Max has taken to build up his exploit - by showing random write into a adjacent application sharing the same gVisor instance. However, he mentions that for the limited write region in this bug:
```

Comment:
i don't really understand the first sentence.  Do you mean this is one of many directions that Max could have taken?

Reply:
Yup!